### PR TITLE
Data transfer pull model

### PR DESCRIPTION
**Pull request by redblom**
Functionality for data transfer based on pull model.

tx-pull-model-01

### CODE REVIEW

**Comment by redblom** [comment]
@labkode @ishank011 please review

**Review by glpatcern — APPROVED** [review]
As discussed with Antoon, this was tested locally and can be merged as soon as https://github.com/cs3org/reva/pull/2052 is ready for merge.

(To be noted that this is a breaking change of the CS3APIs, therefore we have to coordinate the merge of both PRs)

**Comment by labkode** [comment]
@redblom can you bring https://github.com/cs3org/reva/pull/2052 up to date and then we merge it!

File: cs3/gateway/v1beta1/gateway_api.proto [hunk 1]
```diff
@@ -169,6 +170,29 @@ service GatewayAPI {
   // Unsets arbitrary metdata into a storage resource.
   // Arbitrary metadata is returned in a cs3.storage.provider.v1beta1.ResourceInfo.
   rpc UnsetArbitraryMetadata(cs3.storage.provider.v1beta1.UnsetArbitraryMetadataRequest) returns (cs3.storage.provider.v1beta1.UnsetArbitraryMetadataResponse);
+  // Locks a storage resource.
```
Comment (glpatcern) [inline cs3/gateway/v1beta1/gateway_api.proto:173]: @redblom I fear something went wrong with your merge (shouldn't have it been a rebase?), there seem to be duplicated code in the changes - see e.g. this block that comes from a previously merged PR. Can you please have a look?

**Review by glpatcern — APPROVED** [review]
Looks good now following the rebase!

**Pull request merged**